### PR TITLE
Client reduce false positives

### PR DESCRIPTION
**Pull request by Jack-McDowell**

### CODE REVIEW

File: BLUESPAWN-win-client/src/util/eventlogs/EventLogs.cpp [hunk 1]
```diff
@@ -1,294 +1,298 @@
...
+
+        if(GetLastError() != ERROR_NO_MORE_ITEMS) {
+            LOG_VERBOSE(1, "EventLogs::ProcessResults: EvtNext failed with " << GetLastError());
+        }
+
+        return results;
+    }
+
+    std::optional<EventLogItem> EventToEventLogItem(const EventWrapper& hEvent,
+                                                    const std::vector<std::wstring>& params) {
```
Comment (ION28) [inline BLUESPAWN-win-client/src/util/eventlogs/EventLogs.cpp:126]: Codacy found an issue: [Variable 'status' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5932164)

File: BLUESPAWN-win-client/src/util/eventlogs/EventLogs.cpp [hunk 2]
```diff
@@ -1,294 +1,298 @@
...
+                                                &dwBufferUsed);
+                } else {
+                    if(pProperty)
+                        free(pProperty);
+
+                    LOG_ERROR(L"EventLogs::IsChannelOpen: realloc failed for channel " + channel);
+                    return false;
+                }
+            }
+
```
Comment (ION28) [inline BLUESPAWN-win-client/src/util/eventlogs/EventLogs.cpp:254]: Codacy found an issue: [Variable 'status' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5932164)

File: BLUESPAWN-win-client/src/util/eventlogs/EventLogs.cpp [hunk 3]
```diff
@@ -1,294 +1,298 @@
...
+            free(pProperty);
+
+        return pProperty->BooleanVal;
+    }
+
+    bool OpenChannel(const std::wstring& channel) {
+        EVT_HANDLE hChannel = NULL;
+        DWORD status = ERROR_SUCCESS;
+        EVT_VARIANT ChannelProperty;
+        DWORD dwBufferSize = sizeof(EVT_VARIANT);
```
Comment (ION28) [inline BLUESPAWN-win-client/src/util/eventlogs/EventLogs.cpp:271]: Codacy found an issue: [Variable 'dwBufferUsed' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5932164)

File: BLUESPAWN-win-client/src/util/eventlogs/EventLogs.cpp (hunk 1 above)
Comment (ION28) [reply BLUESPAWN-win-client/src/util/eventlogs/EventLogs.cpp:126]: @Jack-McDowell while not important, this finding is valid. seems we set DWORD status = ERROR_SUCCESS a few times within this file, then never use this variable lol

File: BLUESPAWN-win-client/src/util/eventlogs/EventLogs.cpp (hunk 2 above)
Comment (ION28) [reply BLUESPAWN-win-client/src/util/eventlogs/EventLogs.cpp:254]: see above

File: BLUESPAWN-win-client/src/scan/Detection.cpp [hunk 4]
```diff
@@ -21,6 +21,18 @@ size_t ComputeHash(IN CONST std::map<std::wstring, std::wstring>& map) {
     return hash;
 }
 
+size_t ComputeHash(IN CONST std::vector<std::wstring>& values){
+    size_t hash{ 0 };
+
+    std::hash<std::wstring> hasher{};
+    for(auto& val : values){
+        auto first{ hasher(val) };
+        hash = ((hash << 35) | (hash >> 29)) ^ ((first >> 32) | ((first << 32) >> 32));
```
Comment (ION28) [inline BLUESPAWN-win-client/src/scan/Detection.cpp:30]: does this produce any warnings or is there any overflow concerns on x86 @Jack-McDowell ?

Comment (Jack-McDowell) [reply BLUESPAWN-win-client/src/scan/Detection.cpp:30]: Good catch, I'll come up with a fix

**Pull request merged**